src/python_testing/TC_AVSUM_2_1.py

```diff
@@ -0,0 +1,231 @@
+        if self.has_feature_mpan:
+            if mptz.pan is None:
+                asserts.fail("Pan value missing when MPAN feature set")
+            else:
+                apan = mptz.pan
+                asserts.assert_greater_equal(apan, panmin, "Pan out of range of DUT defined values")
+                asserts.assert_less_equal(apan, panmax, "Pan out of range of DUT defined values")
+
+        if self.has_feature_mtilt:
+            if mptz.tilt is None:
+                asserts.fail("Tilt value missing when MTILT feature set")
+            else:
+                atilt = mptz.tilt
+                asserts.assert_greater_equal(atilt, tiltmin, "Tilt out of range of DUT defined values")
+                asserts.assert_less_equal(atilt, tiltmax, "Tilt out of range of DUT defined values")
+
+        if self.has_feature_mzoom:
+            if mptz.zoom is None:
+                asserts.fail("Zoom value missing when MZOOM feature set")
+            else:
+                azoom = mptz.zoom
+                asserts.assert_greater_equal(azoom, 1, "Zoom out of range of DUT defined values")
+                asserts.assert_less_equal(azoom, zoommax, "Zoom out of range of DUT defined values")
+
+    def desc_TC_AVSUM_2_1(self) -> str:
+        return "[TC-AVSUM-2.1] Attributes with DUT as Server"
+
+    def steps_TC_AVSUM_2_1(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Read and verify that one of MTILT, MPAN, or MZOOM is supported"),
+            TestStep(3, "Read and verify ZoomMax attribute, if supported"),
+            TestStep(4, "Read and verify TiltMin attribute, if supported"),
+            TestStep(5, "Read and verify TiltMax attribute, if supported"),
+            TestStep(6, "Read and verify PanMin attribute, if supported"),
+            TestStep(7, "Read and verify PanMax attribute, if supported"),
+            TestStep(8, "Read and verify MPTZPosition attribute."),
+            TestStep(9, "Read and verify MaxPresets attribute, if supported."),
+            TestStep(10, "Read and verify MPTZPresets attribute, if supported."),
+            TestStep(11, "Read and verify DPTZRelativeMove attribute, if supported"),
+        ]
+        return steps
+
+    def pics_TC_AVSUM_2_1(self) -> list[str]:
+        pics = [
+            "AVSUM.S",
+        ]
+        return pics
+
+    @run_if_endpoint_matches(has_cluster(Clusters.CameraAvSettingsUserLevelManagement))
+    async def test_TC_AVSUM_2_1(self):
+        cluster = Clusters.Objects.CameraAvSettingsUserLevelManagement
+        attributes = cluster.Attributes
+        tilt_min_dut = tilt_max_dut = pan_min_dut = pan_max_dut = zoom_max_dut = None
+
+        self.step(1)  # Already done, immediately go to step 2
+
+        feature_map = await self.read_avsum_attribute_expect_success(attribute="FeatureMap")
+        self.has_feature_dptz = (feature_map & cluster.Bitmaps.Feature.kDigitalPTZ) != 0
```

Comment:
does this get used anywhere? I can't find it...should it be used?

Reply:
Line 213!